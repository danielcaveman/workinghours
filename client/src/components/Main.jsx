import React from "react";
import styled from "styled-components";

const StyledContainer = styled.main`
  grid-area: main;
  background-color: #edf0f5;
  padding: 2rem;
  font-size: 1.4rem;
`;

function Main({ children }) {
  return <StyledContainer>{children}</StyledContainer>;
}

export default Main;
