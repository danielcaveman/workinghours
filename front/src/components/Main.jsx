import styled from "styled-components";
import Table from "../components/Table";

const MainContainer = styled.main`
  grid-area: main;
  background-color: #edf0f5;
  padding: 2rem;
  font-size: 1.4rem;
`;

function Main() {
  return (
    <MainContainer>
      <Table />
    </MainContainer>
  );
}

export default Main;
