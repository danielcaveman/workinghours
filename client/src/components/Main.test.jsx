import React from "react";
import { render, screen } from "@testing-library/react";
import Main from "./Main";

describe("<Main />", () => {
  it("Should render main component", () => {
    render(<Main />);
    const main = screen.getByRole("main");

    expect(main).toBeInTheDocument();
  });

  it("Should render main component's children", () => {
    render(
      <Main>
        <div data-testid="children-component"></div>
      </Main>
    );

    const childrenComponent = screen.getByTestId("children-component");
    expect(childrenComponent).toBeInTheDocument();
  });
});
