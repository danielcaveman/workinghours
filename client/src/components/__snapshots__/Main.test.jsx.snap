// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> Should match snapshot 1`] = `
<div>
  <main
    class="sc-bdfBwQ eUeFSZ"
  >
    <div
      data-testid="children-component"
    />
  </main>
</div>
`;
